components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -430,10 +432,17 @@ private OAuth2AccessTokenRespDTO generateNewAccessToken(OAuthTokenReqMessageCont
         return createResponseWithTokenBean(newTokenBean, validityPeriodInMillis, scope);
     }
 
-    private boolean isExistingTokenValid(AccessTokenDO existingTokenBean, long expireTime) {
+    private boolean isExistingTokenValid(OAuthTokenReqMessageContext tokenReqMessageContext,
+            AccessTokenDO existingTokenBean, long expireTime) {
 
         if (TOKEN_STATE_ACTIVE.equals(existingTokenBean.getTokenState()) && expireTime != 0) {
-            return true;
+            Optional<String> selectedACRValue = OAuth2Util.getSelectedACRValue(tokenReqMessageContext);
+            /*
+            * If ACR value is available, check it with the previous ACR value. If ACR value is not available, then
+            * check if the previous access token had an ACR value.
+            * */
+            return selectedACRValue.map(s -> StringUtils.equals(s, existingTokenBean.getAcr()))
```

Comment:
Simplify the IF condition.

Reply:
Here we check the existing token ACR value with the current ACR value, if those are not equal we invalidate the previous token and issue a new token with the new ACR value. Write this stream function with IF condition would be nice since it would give a better understanding